src/thermo/PengRobinsonMFTP.cpp

```diff
@@ -0,0 +1,1321 @@
+    doublereal h_ideal = RT() * mean_X(m_h0_RT);
+    doublereal h_nonideal = hresid();
+	return h_ideal + h_nonideal;
+}
+
+doublereal PengRobinsonMFTP::entropy_mole() const
+{
+    _updateReferenceStateThermo();
+    doublereal sr_ideal = GasConstant * (mean_X(m_s0_R) - sum_xlogx()
+												- std::log(pressure()/refPressure()));
+    doublereal sr_nonideal = sresid();
+	return sr_ideal + sr_nonideal;
+}
+
+doublereal PengRobinsonMFTP::cp_mole() const
+{
+    _updateReferenceStateThermo();
+    doublereal TKelvin = temperature();
+    doublereal sqt2 = sqrt(2);
+    doublereal mv = molarVolume();
+		doublereal vpb = mv + (1 + sqt2)*m_b_current;
+		doublereal vmb = mv + (1 - sqt2)*m_b_current;
+    pressureDerivatives();
+    doublereal cpref = GasConstant * mean_X(m_cp0_R);
+		doublereal dHdT_V = cpref + mv * dpdT_ - GasConstant
+			+ 1.0 / (2.0 * sqt2 *m_b_current) * log(vpb / vmb) * TKelvin *d2aAlpha_dT2();
+		return dHdT_V - (mv + TKelvin * dpdT_ / dpdV_) * dpdT_;
+}
+
+doublereal PengRobinsonMFTP::cv_mole() const
+{
+    _updateReferenceStateThermo();
+    doublereal TKelvin = temperature();
+		doublereal mv = molarVolume();
+		doublereal pp = pressure();
+		pressureDerivatives();
+		doublereal cp = cp_mole();
+		return (cp_mole() + TKelvin* dpdT_* dpdT_ / dpdV_);
+}
+
+doublereal PengRobinsonMFTP::pressure() const
+{
+    _updateReferenceStateThermo();
+    //  Get a copy of the private variables stored in the State object
+    double T = temperature();
+    double mv = meanMolecularWeight() / density();
+		double den = mv * mv + 2 * mv * m_b_current - m_b_current * m_b_current;
+		double pp = GasConstant * T / (mv - m_b_current) - m_aAlpha_current / den;
+    return pp;
+}
+
+void PengRobinsonMFTP::calcDensity()
+{
+    // Calculate the molarVolume of the solution (m**3 kmol-1)
+    const doublereal* const dtmp = moleFractdivMMW();
+    getPartialMolarVolumes(m_tmpV.data());
+    double invDens = dot(m_tmpV.begin(), m_tmpV.end(), dtmp);
+
+    // Set the density in the parent State object directly, by calling the
+    // Phase::setDensity() function.
+    Phase::setDensity(1.0/invDens);
+}
+
+void PengRobinsonMFTP::setTemperature(const doublereal temp)
+{
+    Phase::setTemperature(temp);
+    _updateReferenceStateThermo();
+    updateAB();
+}
+
+void PengRobinsonMFTP::compositionChanged()
+{
+    MixtureFugacityTP::compositionChanged();
+    updateAB();
+}
+
+void PengRobinsonMFTP::getActivityConcentrations(doublereal* c) const
+{
+    getActivityCoefficients(c);
+    for (size_t k = 0; k < m_kk; k++) {
+        c[k] *= moleFraction(k)*pressure()/RT();
+    }
+}
+
+doublereal PengRobinsonMFTP::standardConcentration(size_t k) const
+{
+    getStandardVolumes(m_tmpV.data());
+    return 1.0 / m_tmpV[k];
+}
+
+void PengRobinsonMFTP::getActivityCoefficients(doublereal* ac) const
+{
+    doublereal mv = molarVolume();
+    doublereal T = temperature();
+    doublereal sqt2 = sqrt(2);
+    doublereal vpb_2 = mv + (1 + sqt2)*m_b_current;
```

Comment:
Frankly, for no specific reason! But I have changed it now to 'vpb2' and so on.

Reply:
But why is there still a `2` in the name?

Reply:
Typically, vpb and vmb are used to represent (v+b) and (v-b). Here vpb2 is used to avoid the confusion since it represents [v + (1 +sqrt(2)) * b].

Reply:
I'm not sure of the convention used in Cantera, but cryptic variable names always worry me when I see them.

This resource may be useful: https://dzone.com/articles/naming-conventions-from-uncle-bobs-clean-code-phil

Reply:
In this case, I think the variable name is about as good as its going to get.  

1. There is no real physical meaning which would lead to a meaningful name.  It's just a kluge-y function that we can clean the code a bit by defining it separately.
2. We could in theory change `vpb21ecause `vpb` and `vmb` are used repeatedly throughout  this file and refer specifically to `molarvolume + b` and `molarvolume - b`, respectively, I think simplifying the name in this case would actually lead to confusion.
3. The only alternate I think would be to scrap these as named variables and just write out the equation in full on line 224.  The variables are each used only once, so it wouldn't be _too_ bad...